core/src/main/java/com/google/errorprone/bugpatterns/UngroupedOverloads.java

```diff
@@ -96,7 +96,9 @@ public Description matchClass(ClassTree classTree, VisitorState state) {
       Tree member = classTree.getMembers().get(i);
       if (member instanceof MethodTree) {
         MethodTree methodTree = (MethodTree) member;
-        methods.put(OverloadKey.create(methodTree), MemberWithIndex.create(i, methodTree));
+        if (!ASTHelpers.isGeneratedConstructor(methodTree)) {
+          methods.put(OverloadKey.create(methodTree), MemberWithIndex.create(i, methodTree));
+        }
```

Comment:
This check matches the [constraint](https://github.com/google/error-prone/blob/1d23141bd74663ad12e20534d062bef223df6efa/check_api/src/main/java/com/google/errorprone/fixes/SuggestedFix.java#L351-L359) that otherwise triggers the `IllegalArgumentException` reported in #3404. As these generated constructors are invisible, it makes sense (I suppose) that `UngroupedOverloads` shouldn't care about their location.